app/services/bulk_upload/lettings/row_parser.rb

```diff
@@ -804,6 +840,30 @@ def ethnic_group_from_ethnic
     end
   end
 
+  def brent
+    if %i[field_80 field_81 field_82 field_83].any? { |f| public_send(f).present? }
+      self.field_80 ||= 0
```

Comment:
I wrote this logic (and the equivalent logic for `scharge`, `pscharge` and `supcharg`) based on the derived variable logic in `app/models/derived_variables/lettings_log_variables.rb:23`. Would be interested to see if it could be done in a better way.

Reply:
I don't think we need the if statement, because we assume that these logs are completed, so the values in `field_80 field_81 field_82 field_83` should be what they're meant to be. Except if it's a care home, then these won't get set to 0 so they'd look like they're different...
What If you create a log in memory and run .valid? on it? And then use the fields from that log to check if an old log exists. That way all the values would get inferred and you wouldn't need to do it yourself

Reply:
Hey @kosiakkatrina, thanks for this. I've tried implementing your idea doing something like the following: 

`
field_mappings = {
  "startdate" => startdate,
  "postcode_full" => postcode_full,
  "brent" => field_80,
  "scharge" => field_81,
  "pscharge" => field_82,
  "supcharg" => field_83,
  "tenancycode" => field_7,
  "age1" => field_12,
  "sex1" => field_20,
  "ecstat1" => field_35,
  "ethnic" => field_43,
}
log = LettingsLog.new(field_mappings)
log.valid?
field_mappings.update(field_mappings) { |key, value| log[key] } # I found this online as a way to easily reassign the values of the hash
duplicate_log_exists = LettingsLog.exists?(field_mappings)
`

However I'm having trouble - calling `log.valid?` seems to change `brent`, `scharge` etc from non-nil values to nil, and I'm not sure why. I thought it might be due to `app/models/lettings_log.rb#validate_rent_range` but this isn't getting triggered. Is it because I haven't filled in some essential fields for the log I'm creating in memory? If so, wouldn't I need to give values for almost _all_ the fields for the log I'm creating? (I tried to find which fields would be essential, and it seems to be most of them based on `app/models/lettings_log.rb#optional_fields`.) Specifying values for almost all the fields for the log I'm creating in memory doesn't feel like the right approach, but it might be!